build-system/compile/single-pass.js

```diff
@@ -538,7 +541,7 @@ exports.singlePassCompile = function(entryModule, options) {
  * magic-string might be part of the solution here so explore that (pre or post
  * process)
  */
-function wrapMainBinaries() {
+async function wrapMainBinaries() {
```

Comment:
I had to make most of the functions that are directly used in `gulp` tasks `async` to prevent the case where the outer task would immediately signal completion while the actual function was still running.

However, you're right that there are a few functions like `wrapMainBinaries`, `postProcessConcat`, and two types of `compile` that don't need to be `async`. I've reverted them.